configs/api/api.go

```diff
@@ -50,10 +50,18 @@ func (a *API) RegisterRoutes(r *mux.Router) {
 		handler            http.HandlerFunc
 	}{
 		{"root", "GET", "/", a.admin},
+		// Deprecated public APIs
 		{"get_config", "GET", "/api/configs/org/cortex", a.getConfig},
 		{"set_config", "POST", "/api/configs/org/cortex", a.setConfig},
+		// Dedicated APIs for updating rules config. In future, these *must*
+		// be used.
+		{"get_rules", "GET", "/api/prom/rules", a.getConfig},
+		{"set_rules", "POST", "/api/prom/rules", a.setConfig},
+		{"get_alert_config", "GET", "/api/prom/alerts", a.getConfig},
```

Comment:
Since both the Ruler and the Alertmanager do stuff with alerts and this is specifically about the Alertmanager's config, maybe name this more explicitly:

`{"get_alertmanager_config", "GET", "/api/prom/config/alertmanager", a.getConfig},`

And rules also under `/api/prom/config/rules`? Or something like that.